lib/library.rb

```diff
@@ -0,0 +1,45 @@
+class Library
+  attr_reader :authors, :books, :name, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @authors = []
+    @books = []
+    @checked_out_books = []
+    @books_by_popularity = Hash.new(0)
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.map {|book| @books << book}
+  end
+
+  def publication_time_frame_for(author)
+    {start: author.first_book_published, end: author.last_book_published}
```

Comment:
I like that you built helper methods for this one. I'm hoping I find tests for those helper methods!